tests/conftest.py

```diff
@@ -43,6 +125,10 @@ def test_app(make_app, request):
         docutilsconf=builder_params.get("docutilsconf"),
         parallel=builder_params.get("parallel", 0),
     )
+    # Add the spec_pattern as an attribute to the Sphinx app object
+    app.spec_pattern = builder_params.get("spec_pattern", "")
+    # Add the test_js() function as an attribute to the Sphinx app object
+    app.test_js = test_js.__get__(app, Sphinx)
```

Comment:
Ok, the Python gods have left me.
You need to explain this dark voodoo in a short call. Looks interesting :)

Reply:
Sure.